### PR TITLE
IDE-2668 gradle dependence quick fix

### PR DESCRIPTION
**Pull request by lovettli**

### CODE REVIEW

**Comment by gamerson** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/parser/GradleDependencyUpdater.java [hunk 1]
```diff
@@ -37,19 +38,27 @@ public GradleDependencyUpdater( String scriptContents ) throws MultipleCompilati
         nodes = builder.buildFromString( scriptContents );
     }
 
+    public FindDependenciesVisitor insertDependency( GradleDependency gradleDependency ) throws IOException
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/parser/GradleDependencyUpdater.java]: What happens if there is no "dependencies {}" block at all?  Does this work?  Remember the "dependencies" call could happen in a parent build.gradle.  Do we have units tests for "insertDependency" for when there is no "dependencies" code block at all?

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    }
+
+    private void createDepProposal(
+        IInvocationContext context, Collection<IJavaCompletionProposal> proposals, ServiceContainer bundle )
+    {
+        final String bundleName = bundle.getBundleName();
+        final String bundleVersion = bundle.getBundleVersion();
+
+        proposals.add(
+            new CUCorrectionProposal( "Add Liferay Dependence " + bundleName, context.getCompilationUnit(), null, -0 )
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java]: Call it "Add Gradle dependency" don't mention "Liferay"

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                @Override
+                public void apply( IDocument document )
+                {
+                    try
+                    {
+                        GradleDependencyUpdater updater = new GradleDependencyUpdater( gradleFile.getLocation().toFile() );
+                        List<GradleDependency> existDependencies = updater.getAllDependencies();
+                        GradleDependency gd;
+
+                        if( bundleName.equals( "com.liferay.portal.kernel" ) )
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java]: We need to add supprt for other dependencies that are very common like:
`org.osgi`, `javax.portlet`, and `javax.servlet`.  any dependency that is used in blade-samples or in project templates

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+                    catch( Exception e )
+                    {
+                        GradleCore.logError( "Gradle dependence got error", e );
+                    }
+                }
+
+                @Override
+                public Object getAdditionalProposalInfo( IProgressMonitor monitor )
+                {
+                    return "Add dependenece " + bundleName + " " + bundleVersion;
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java]: `Add dependency`

File: tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+import com.liferay.ide.gradle.core.GradleCore;
+import com.liferay.ide.gradle.core.parser.GradleDependency;
+import com.liferay.ide.gradle.core.parser.GradleDependencyUpdater;
+import com.liferay.ide.project.core.modules.ServiceContainer;
+import com.liferay.ide.project.core.util.TargetPlatformUtil;
+
+/**
+ * @author Lovett Li
+ */
+public class QuickFixGradleDep implements IQuickFixProcessor
```
Comment (gamerson) [inline tools/plugins/com.liferay.ide.gradle.ui/src/com/liferay/ide/gradle/ui/quickfix/QuickFixGradleDep.java]: The Icon for this should be the same `Gradle` icon that is in buildship

**Comment by gamerson** [comment]
Please resend, thanks! <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by gamerson** [comment]
screen shot 2016-08-08 at 10 32 58 pm

We need to support quick fix for these dependencies

File: tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/parser/GradleDependencyUpdater.java (hunk 1 above)
Comment (lovettli) [inline tools/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/parser/GradleDependencyUpdater.java]: User click quick fix then we will check build.gradle file in current project then insert dependence.  Yes. we have junit case test for no dependencies code block. we will add a new dependencies block if there is no dependencies.

**Comment by lovettli** [comment]
Yes.It is very necessary to add these dependencies are related by our project. I am not sure how can we get information of these classes which are not belong to liferay service. We can obtain bundle name bundle version from target platform but for these class are indirect referenced by portal.kernel. question is how can we get group , name and version ?

**Comment by gamerson** [comment]
We will just have to build up our own internal database of these packages and what suggestions we can make to fix this.  We can just look at our blade samples for packages and the dependencies that we can suggest to fix them.  Like

`import javax.servlet.*` should be `compile 'javax.servlet:javax.servlet-api:3.0.1'` etc

**Pull request closed without merging**